airflow-core/src/airflow/api_fastapi/core_api/routes/public/hitl_shared_links.py

```diff
@@ -0,0 +1,293 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from json import JSONDecodeError
+
+import structlog
+from cryptography.exceptions import InvalidTag
+from fastapi import Depends, HTTPException, Request, status
+from fastapi.responses import RedirectResponse
+from pydantic import HttpUrl
+
+from airflow.api_fastapi.auth.managers.models.resource_details import DagAccessEntity
+from airflow.api_fastapi.common.db.common import SessionDep
+from airflow.api_fastapi.common.router import AirflowRouter
+from airflow.api_fastapi.core_api.datamodels.hitl import HITLDetailResponse
+from airflow.api_fastapi.core_api.datamodels.hitl_shared_links import (
+    GenerateSharedLinkRequest,
+    SharedLinkResponse,
+)
+from airflow.api_fastapi.core_api.openapi.exceptions import create_openapi_http_exception_doc
+from airflow.api_fastapi.core_api.security import GetUserDep, requires_access_dag
+from airflow.api_fastapi.core_api.services.public.hitl_shared_links import (
+    _generate_hitl_shared_link,
+    _retrieve_hitl_redirect_url,
+    _update_hitl_detail_through_token,
```

Comment:
I think the "privately marked" methods should be converted to be "non private" to be used.

Reply:
sounds good

Reply:
done